test/acceptance/user-order.controller.acceptance.ts

```diff
@@ -0,0 +1,199 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {createClientForHandler, supertest, expect} from '@loopback/testlab';
+import {RestServer} from '@loopback/rest';
+import {ShoppingApplication} from '../..';
+import {OrderRepository, UserRepository} from '../../src/repositories';
+import {MongoDataSource} from '../../src/datasources';
+import {User, Order, ShoppingCartItem} from '../../src/models';
+
+describe('UserOrderController acceptance tests', () => {
+  let app: ShoppingApplication;
+  let server: RestServer;
+  let client: supertest.SuperTest<supertest.Test>;
+  const orderRepo = new OrderRepository(new MongoDataSource());
+  const userRepo = new UserRepository(new MongoDataSource(), orderRepo);
+
+  before(setupApplication);
+  beforeEach(clearDatabase);
+  after(async () => {
+    await app.stop();
+  });
+
+  it('creates an order for a user with a given orderId', async () => {
+    const user = await givenAUser();
+    const userId = user.id.toString();
+    const order = {
```

Comment:
Have you considered using a test-data builder here and specify only few property values that are relevant for this test?

Acceptance tests are verifying that all application parts work together, they should not be verifying behavior of individual components (we have unit and integration tests for that). 

In this particular case, we should rely on integration tests to verify that all properties provided to `orderRepo.create` are correctly stored in the database and retrieved back. IMO, we should use `const order = givenAnOrder()` here, because the actual order details are not important for this test.